File: Ruby/lambdaconscarcdr.rb
#!/usr/bin/env ruby

Kons  = ->(hd, tl=nil) { -> x { if x then hd else tl end }}
Virst = -> l { l.(true ) }
Rrest = -> l { l.(false) }

Lstt  = Kons.(1, Kons.(2))

Virst.(Rrest.(Lstt))

File: C#/lambdaconscarcdr.cs
class LambdaConsCarCdrCSharp
{
    static dynamic LambdaConsCarCdr()
    {
        System.Func<dynamic, dynamic> kons(dynamic hd, dynamic tl) => (dynamic x) => x ? hd : tl;

        dynamic virst(dynamic l) => l(true);

        dynamic rrest(dynamic l) => l(false);

        var lstt = kons(1, kons(2, null));

        return virst(rrest(lstt));
    }

    public static void Main()
    {
        LambdaConsCarCdr();
    }
}
